cli/src/deploy/mod.rs

```diff
@@ -445,22 +445,17 @@ pub async fn add_deploy_config() {
     .unwrap();
   let mut cloud_configs = Vec::new();
   let options = creds.keys().cloned().collect::<Vec<String>>();
+  if options.len() == 0 {
+    prompt_new_cred(false).await
+  }
   loop {
-    let cred = if creds.len() > 1 {
-      let selection = Select::new()
-        .items(&options)
-        .with_prompt("Pick Credential to use")
-        .default(0)
-        .interact()
-        .unwrap();
-      Some(options[selection].to_string())
-    } else {
-      if creds.len() == 0 {
-        prompt_new_cred(false).await
-      }
-      // use default, or only, credential
-      None
-    };
+    let selection = Select::new()
+      .items(&options)
+      .with_prompt("Pick Credential to use")
+      .default(0)
+      .interact()
+      .unwrap();
```

Comment:
Do we want to have a special "Create new credentials" field that they can select that triggers the `prompt_new_cred` path? you have above, as well?

We could also simplify it by always adding that to the list, so the `options.len()` is always >=1 and that's the only option only if there are no credentials already present?

Reply:
Good catch. I will generate `options` after calling `prompt_new_cred`